test/use-swr-updated-at.test.tsx

```diff
@@ -0,0 +1,222 @@
+import { act, fireEvent, screen } from '@testing-library/react'
+import React, { useEffect, useReducer } from 'react'
+import useSWR from 'swr'
+import { createKey, renderWithConfig, sleep } from './utils'
+
+describe('useSWR - updatedAt', () => {
+  it('should be initially undefined', async () => {
+    const key = createKey()
+
+    const fetcher = () => {
+      return 'data'
+    }
+
+    function Page() {
+      const { mutate, updatedAt } = useSWR(key, fetcher, {
+        revalidateOnMount: false
+      })
+
+      return <button onClick={() => mutate()}>data: {updatedAt}</button>
+    }
+
+    renderWithConfig(<Page />)
+
+    screen.getByText('data:')
+  })
+
+  it('should not trigger re-render if not consumed', async () => {
+    const key = createKey()
+
+    const fetcher = () => {
+      return 'data'
+    }
+
+    const renderSpy = jest.fn()
+
+    function Page() {
+      const { mutate } = useSWR(key, fetcher, {
+        revalidateOnMount: false
+      })
+
+      renderSpy()
+
+      return <button onClick={() => mutate()}>data</button>
+    }
+
+    renderWithConfig(<Page />)
+
+    screen.getByText('data')
+
+    fireEvent.click(screen.getByRole('button'))
+
+    await act(async () => {
+      await new Promise<void>(resolve => window.queueMicrotask(resolve))
+    })
+
+    expect(renderSpy).toHaveBeenCalledTimes(1)
+  })
+
+  it('should eventually reflect the last time the fetcher was called', async () => {
+    const key = createKey()
+
+    let fetcherCallTime: number
+
+    const fetcher = () => {
+      fetcherCallTime = Date.now()
+      return 'data'
+    }
+
+    const updateSpy = jest.fn()
+
+    function Page() {
+      const { mutate, updatedAt } = useSWR(key, fetcher)
+
+      useEffect(() => {
+        updateSpy(updatedAt)
+      }, [updatedAt])
+
+      return <button onClick={() => mutate()}>data</button>
+    }
+
+    renderWithConfig(<Page />)
+
+    screen.getByText('data')
+
+    fireEvent.click(screen.getByRole('button'))
+
+    expect(updateSpy).toHaveBeenCalledTimes(1)
+
+    expect(updateSpy.mock.calls[0][0]).toBeUndefined()
+
+    await act(async () => {
+      await new Promise<void>(resolve => window.queueMicrotask(resolve))
+    })
+
+    expect(updateSpy).toHaveBeenCalledTimes(2)
+
+    const updatedAt = updateSpy.mock.calls[1][0]
+    expect(updatedAt).toBeDefined()
+
+    expect(updatedAt).toBeGreaterThanOrEqual(fetcherCallTime)
+  })
+
+  it('should be consistent in all hooks using the same key', async () => {
```

Comment:
Rather long test.

- Four hooks using the same key are at play
- Initially, Page (zero), a Dashboard mock (first), and a mutator are present.
- During initial rendering, `updatedAt` is undefined.
- Eventually, call mutate from Page.
- Expect Page and Dashboard to agree on `updatedAt`
- I am not 100% sure why, but after this I ought to wait on the deduping interval to settle
- After that, mount the second dashboard, configured to revalidate on mount
- Assert that `updatedAt` for `second` is undefined
- Eventually `second` agrees with `zero`
- Trigger an update from the `mutator`
- Values of `updatedAt` should increase
-All hooks should agree on `updatedAt`